File: core/src/com/gofficer/codenames/NetworkDictionary.java
package com.gofficer.codenames;

import com.artemis.Component;
import com.gofficer.codenames.components.*;
import com.gofficer.codenames.network.notification.EntityUpdate;
import net.mostlyoriginal.api.network.marshal.common.MarshalDictionary;

import java.awt.*;
import java.util.HashMap;
import java.util.HashSet;
import java.util.concurrent.ConcurrentHashMap;

public class NetworkDictionary extends MarshalDictionary {

    public NetworkDictionary() {
        registerAll(

                // Game Notifications
                EntityUpdate.class,
//                EntityUpdate.EntityUpdateBuilder.class,
                // Other
                boolean[][].class,
                boolean[].class,
                int[][].class,
                int[].class,
                Integer[].class,
                ConcurrentHashMap.class,
                HashMap.class,
                HashSet.class,
                Component.class,
                Component[].class,
                Class.class,
                Class[].class, // TODO not sure how to do this in kotlin hence the java file
                Object.class,
                Network.class,
                Network.Client.JoinRoomRequest.class,
                EntityUpdate.EntityUpdateBuilder.class,

                PositionComponent.class,
                CardComponent.class,
                NetworkComponent.class,
                RevealedComponent.class,
                TextureReferenceComponent.class
        );
    }

    private void registerAll(Class... classes) {
        topId = 40;
        for (Class clazz : classes) {
            register(topId++, clazz);
        }
    }
}

File: core/src/com/gofficer/codenames/NetworkDictionary.kt
package com.gofficer.codenames

import com.artemis.Component
import net.mostlyoriginal.api.network.marshal.common.MarshalDictionary
import java.util.*
import java.util.concurrent.ConcurrentHashMap
import java.util.concurrent.ConcurrentLinkedDeque


class NetworkDictionary : MarshalDictionary() {
    init {
        registerAll(



            // Game Requests
            Array<Any>::class.java,
            // Other
            Array<BooleanArray>::class.java,
            BooleanArray::class.java,
            Array<IntArray>::class.java,
            IntArray::class.java,
            Array<Int>::class.java,
            ConcurrentHashMap::class.java,
            HashMap::class.java,
            HashSet::class.java,
            ConcurrentLinkedDeque::class.java,
            Component::class.java,
            Array<Component>::class.java,
            Class::class.java,
            Array<Any>::class.java,
            Map::class.java,
            Optional::class.java,
            Any::class.java,
            Network::class.java
        )
    }

    private fun registerAll(vararg classes: Class<*>) {
        topId = 40
        for (clazz in classes) {
            register(topId++, clazz)
        }
    }
}
